indico/modules/rb/client/js/common/bookings/BookingDetails.jsx

```diff
@@ -487,21 +508,58 @@ class BookingDetails extends React.Component {
               disabled={bookingStateChangeInProgress}
               content={Translate.string('Cancel booking')}
             />
-            <Confirm
-              header={Translate.string('Confirm cancellation')}
-              content={Translate.string(
-                'Are you sure you want to cancel this booking? ' +
-                  'This will cancel future occurrences of this booking.'
-              )}
-              confirmButton={<Button content={Translate.string('Cancel booking')} negative />}
-              cancelButton={Translate.string('Close')}
+            <Modal
               open={activeConfirmation === 'cancel'}
-              onCancel={this.hideConfirm}
-              onConfirm={() => {
-                this.changeState('cancel');
-                this.hideConfirm();
-              }}
-            />
+              size="small"
+              onClose={() => this.hideConfirm()}
+            >
+              <Modal.Header>
+                <Translate>Confirm cancellation</Translate>
+              </Modal.Header>
+              <Modal.Content>
+                <p>
+                  <PluralTranslate count={occurrenceCount}>
+                    <Singular>Are you sure you want to cancel this booking?</Singular>
+                    <Plural>
+                      Are you sure you want to cancel this booking? This will cancel all{' '}
+                      <Param name="count" value={occurrenceCount} wrapper={<strong />} />{' '}
+                      occurrences.
+                    </Plural>
+                  </PluralTranslate>
+                </p>
+                <p>
+                  {occurrenceCount > 1 && (
+                    <Translate>
+                      Single occurrences can be cancelled via the timeline view.
+                    </Translate>
+                  )}
+                </p>
+              </Modal.Content>
+              <Modal.Actions>
+                <Button onClick={this.hideConfirm} content={Translate.string('Close')} />
+                {occurrenceCount > 1 && (
+                  <Button
+                    icon="calendar outline"
+                    onClick={() => {
+                      this.hideConfirm();
+                      this.showOccurrences();
+                    }}
+                    content={Translate.string('View Timeline')}
+                  />
+                )}
+                <Button
+                  onClick={() => {
+                    this.hideConfirm();
+                  }}
+                  content={PluralTranslate.string(
+                    'Cancel booking',
+                    'Cancel all occurrences',
+                    occurrenceCount
+                  )}
```

Comment:
I think using separate Translates (with the `?:` conditional) would be slightly more correct here, since the translation does not depend on the number, just on the fact whether it's a single occurrence or not.